client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDriverUri.java

```diff
@@ -357,6 +362,18 @@ public void setupClient(OkHttpClient.Builder builder)
         }
     }
 
+    private static DnsResolver instantiateDnsResolver(Class<? extends DnsResolver> resolverClass, Optional<String> context)
+    {
+        try {
+            return context.isPresent() ?
```

Comment:
Let's simplify and require the single argument constructor. Using a different method signature based on whether or not the user provided a context argument seems strange. We can pass null if the context is not present.